c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -419,15 +529,23 @@ class CachingAllocatorConfig {
                 "garbage_collect_threshold too big, set it 0.0~1.0",
                 "");
             m_garbage_collection_threshold = val2;
+            used_native_specific_option = true;
           } else {
             TORCH_CHECK(false, "Unrecognized CachingAllocator option: ", kv[0]);
           }
         }
       }
+
+      if (used_cudaMallocAsync && used_native_specific_option) {
+        TORCH_WARN("backend:cudaMallocAsync ignores max_split_size_mb, "
+                   "roundup_power2_divisions, and garbage_collect_threshold.");
+      }
     }
   }
 };
 
+namespace Native {
```

Comment:
fwiw capitalized namespace is pretty unusual, it will make people confused if they're looking at a class as opposed to a namespace